UiPath.FreeRdpClient/UiPath.SessionTools.Tests/ProcessRunnerExtensionsTests.cs

```diff
@@ -0,0 +1,104 @@
+    }
+
+    [Fact(DisplayName = $"{nameof(Sut.SuppressCancellation)} should return a Task that succeeds as soon as the provided Task succeeds.")]
+    public async Task SuppressCancellation_ShouldReturnATaskThatSucceeds_WhenTheProvidedTaskSucceeds()
+    {
+        TaskCompletionSource source = new();
+        var task = source.Task.SuppressCancellation();
+
+        task.IsCompletedSuccessfully.ShouldBeFalse();
+
+        source.SetResult();
+
+        Should.CompleteIn(
+            task.ShouldNotThrowAsync(),
+            TimeSpan.FromMilliseconds(100));
+    }
+
+    [Fact(DisplayName = $"{nameof(Sut.SuppressCancellation)} should return a Task that fails as soon as the provided Task fails.")]
+    public async Task SuppressCancellation_ShouldReturnATaskThatFails_WhenTheProvidedTaskFails()
+    {
+        DivideByZeroException ex = new();
+
+        TaskCompletionSource source = new();
+        var task = source.Task.SuppressCancellation();
+
+        task.IsCompletedSuccessfully.ShouldBeFalse();
+
+        source.SetException(ex);
+
+        Should.CompleteIn(
+            task.ShouldThrowAsync<DivideByZeroException>(),
+            TimeSpan.FromMilliseconds(100)
+        ).ShouldBe(ex);
+    }
+
+
+
+    [Theory(DisplayName = $"{nameof(Sut.ThrowIfNonZeroCode)} should not throw when the provided exit code is 0 or null.")]
+    [InlineData(0)]
+    [InlineData(null)]
+    public void ThrowIfNonZeroCode_ShouldNotThrow_WhenTheProvidedExitCodeIs0OrNull(int? exitCode)
+    {
+        var report = _fixture.MockReport(exitCode);
+
+        var act = () => report.ThrowIfNonZeroCode();
+
+        act.ShouldNotThrow();
+    }
+
+    [Fact(DisplayName = $"{nameof(Sut.ThrowIfNonZeroCode)} should throw when the provided exit code is not 0 nor null.")]
+    public void ThrowIfNonZeroCode_ShouldThrow_WhenTheProvidedExitCodeIsNot0NorNull()
+    {
+        var report = _fixture.MockReport(1);
+
+        var act = () => report.ThrowIfNonZeroCode();
+
+        var ex = act.ShouldThrow<ProcessRunner.NonZeroExitCodeException>();
+    }
+
+    [Theory(DisplayName = $"{nameof(Sut.ThrowIfNonZeroCode)} (async) should not throw when the provided exit code is 0 or null.")]
+    [InlineData(0)]
+    [InlineData(null)]
+    public async Task AsyncThrowIfNonZeroCode_ShouldNotThrow_WhenTheProvidedExitCodeIs0OrNull(int? exitCode)
+    {
+        var taskReport = _fixture.Shortly(_fixture.MockReport(exitCode));
+
+        var asyncAct = taskReport.ThrowIfNonZeroCode;
+
+        await asyncAct.ShouldNotThrowAsync();
+    }
+
+    [Fact(DisplayName = $"{nameof(Sut.ThrowIfNonZeroCode)} (async) should throw when the provided exit code is not 0 nor null.")]
+    public async Task AsyncThrowIfNonZeroCode_ShouldThrow_WhenTheProvidedExitCodeIsNot0NorNull()
+    {
+        var taskReport = _fixture.Shortly(_fixture.MockReport(exitCode: 1));
+
+        var asyncAct = taskReport.ThrowIfNonZeroCode;
+
+        await asyncAct.ShouldThrowAsync<ProcessRunner.NonZeroExitCodeException>();
```

Comment:
it would look nicer when it fails if you inline the actual expression instead of the `asyncAct`